operator/src/main/java/oracle/kubernetes/operator/PvcWatcher.java

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2023, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import io.kubernetes.client.openapi.models.V1ObjectMeta;
+import io.kubernetes.client.openapi.models.V1PersistentVolumeClaim;
+import io.kubernetes.client.openapi.models.V1PersistentVolumeClaimStatus;
+import oracle.kubernetes.common.logging.MessageKeys;
+import oracle.kubernetes.operator.calls.CallResponse;
+import oracle.kubernetes.operator.helpers.CallBuilder;
+import oracle.kubernetes.operator.helpers.DomainPresenceInfo;
+import oracle.kubernetes.operator.helpers.ResponseStep;
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.operator.steps.DefaultResponseStep;
+import oracle.kubernetes.operator.work.NextAction;
+import oracle.kubernetes.operator.work.Packet;
+import oracle.kubernetes.operator.work.Step;
+
+/** Watches for pvcs to become Ready or leave Ready state. */
+public class PvcWatcher implements PvcAwaiterStepFactory {
+
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Operator", "Operator");
+
+  // Map of Pvc name to Runnable
+  private final Map<String,Consumer<V1PersistentVolumeClaim>> completeCallbackRegistrations = new ConcurrentHashMap<>();
+  public final DomainProcessor processor;
+
+  PvcWatcher(DomainProcessor processor) {
+    this.processor = processor;
+  }
+
+  /**
+   * Test if PersistentVolumeClaim is bound.
+   * @param pvc PersistentVolumeClaim
+   * @return true, if bound
+   */
+  public static boolean isBound(V1PersistentVolumeClaim pvc) {
+    if (pvc == null) {
+      return false;
+    }
+
+    V1PersistentVolumeClaimStatus status = pvc.getStatus();
+    LOGGER.fine("Status of pvc " + Objects.requireNonNull(pvc.getMetadata()).getName() + ": " + status);
+    if (status != null) {
+      String phase = status.getPhase();
```

Comment:
Should we print the phase value?  I am not sure if there is other phase value that we may encounter and handle